### PR TITLE
Add CMake support

### LINKED ISSUE #211: Switch to CMake
Outstanding TODOs/tests:

- [x] Package CSL2XSB
- [x] Test CI build and installer creation for all platform
- [x]  Package ``vcredist`` on Windows
- [x] Split debug symbols from executable/dylibs on macOS
- [x] Package ``create_dbus_machineid.scpt`` on macOS
- [x]  Build  and package MS Simulator plugins on Windows

### PR DESCRIPTION
**Pull request by ltoenning**
Fixes #211 

:warning: Before going into a release version, this should be tested by some users beforehand.

### CODE REVIEW

File: scripts/build.py [hunk 1]
```diff
@@ -66,22 +72,51 @@ def checks(self):
         build_path = self._get_swift_build_path()
         os.chdir(build_path)
         if self._should_run_checks():
-            subprocess.check_call([self.make_cmd, 'check', '-j1'], env=dict(os.environ))
+            subprocess.check_call(["cmake", "--build", ".", "--target", "test"], env=dict(os.environ))
```
Comment (oktal3700) [inline scripts/build.py:75]: How about `subprocess.check_call(["ctest", "--verbose"]` so we see the full test output on CI

**Review by oktal3700 — CHANGES_REQUESTED** [review]
- [ ] I guess you intended to rename the unittests to remove the `black` prefix
- [ ] There is one test which passes with qmake but is skipped with cmake: `BlackCoreTest::CTestContext`

File: .gitignore
Comment (oktal3700) [inline .gitignore:1]: Lines 5, 36, 37 & 54 can be removed. Those files were artifacts of qmake.

File: .gitignore [hunk 2]
```diff
@@ -55,3 +56,7 @@ desktop.ini
 # Jetbrains stuff
 .idea/**
 cmake-build-*/
+
+# Install folders
+out/
+dist/
```
Comment (oktal3700) [inline .gitignore:62]: These two directories are created only in the root of the source tree, right? So they can be prepended with a `/` prefix to limit their scope there:
```
/out/
/dist/
```

File: default.json
Comment (oktal3700) [inline default.json:1]: Could be renamed to `version.json`

File: scripts/build.py [hunk 3]
```diff
@@ -265,33 +310,19 @@ def _specific_prepare(self):
...
-    def _should_run_checks(self):
-        return False
+    def _strip_debug(self):
+        pass
 
-    def _should_publish(self):
-        return True
+    def bundle_csl2xsb(self):
+        if self.word_size != '64':
+            return
```
Comment (oktal3700) [inline scripts/build.py:321]: It makes sense to bundle CSL2XSB for both 32bit and 64bit Windows. Both can be used with X-Plane (with 64bit xswiftbus, of course)

File: src/plugins/simulator/fs9/lobbyclient.h
Comment (oktal3700) [inline src/plugins/simulator/fs9/lobbyclient.h:1]: Prefer putting this into a separate commit

File: src/plugins/simulator/fsxcommon/simconnectwindows.h [hunk 4]
```diff
@@ -18,6 +18,8 @@
 #include <Windows.h>
 #include <SimConnect.h>
 
+#include <QtCore/QtGlobal>
```
Comment (oktal3700) [inline src/plugins/simulator/fsxcommon/simconnectwindows.h:21]: Should be `#include <QtGlobal>`

File: tests/blackmisc/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+        NAME blackmisc_propertyindex
+        SOURCES testpropertyindex/testpropertyindex.cpp
+        LINK_LIBRARIES misc tests_test Qt::Core
+)
+
+#add_swift_test(
+#        NAME blackmisc_sharedstate
+#        SOURCES testsharedstate/testsharedstate.cpp
+#        LINK_LIBRARIES misc tests_test Qt::Core
+#)
```
Comment (oktal3700) [inline tests/blackmisc/CMakeLists.txt:157]: Commented out unittest?

File: .gitignore
Comment (oktal3700) [inline .gitignore:1]: Seems like this should be squashed into a9abbf4

**Comment by ltoenning** [comment]
Merged with staging branch. https://github.com/swift-project/pilotclient/commit/1deaeba58431f3879e35cdc5604da5944a3c919f
Fixed #220.

**Pull request closed without merging**